diff -Naur gnome-screensaver-3.6.0.orig/src/gnome-screensaver.desktop.in gnome-screensaver-3.6.0/src/gnome-screensaver.desktop.in
--- gnome-screensaver-3.6.0.orig/src/gnome-screensaver.desktop.in	2013-09-22 20:54:14.251604475 +0700
+++ gnome-screensaver-3.6.0/src/gnome-screensaver.desktop.in	2013-09-22 21:04:32.518273637 +0700
@@ -5,7 +5,7 @@
 Icon=preferences-desktop-screensaver
 Exec=gnome-screensaver
 OnlyShowIn=GNOME;
-AutostartCondition=GNOME3 if-session gnome-fallback
+AutostartCondition=GNOME3 unless-session gnome
 NoDisplay=true
 X-GNOME-Autostart-Delay=20
 X-GNOME-Autostart-Phase=Application
diff -Naur gnome-screensaver-3.6.0.orig/src/gnome-screensaver.desktop.in.in gnome-screensaver-3.6.0/src/gnome-screensaver.desktop.in.in
--- gnome-screensaver-3.6.0.orig/src/gnome-screensaver.desktop.in.in	2013-09-22 20:54:14.251604475 +0700
+++ gnome-screensaver-3.6.0/src/gnome-screensaver.desktop.in.in	2013-09-22 21:04:52.690273114 +0700
@@ -5,7 +5,7 @@
 Icon=preferences-desktop-screensaver
 Exec=gnome-screensaver
 OnlyShowIn=GNOME;
-AutostartCondition=GNOME3 if-session gnome-fallback
+AutostartCondition=GNOME3 unless-session gnome
 NoDisplay=true
 X-GNOME-Autostart-Delay=20
 X-GNOME-Autostart-Phase=Application
